app/controllers/pages_controller.rb

```diff
@@ -5,7 +5,24 @@ class PagesController < ApplicationController
   skip_before_action :authenticate_user!
   skip_authorization_check
 
-  def home; end
+  def home
+    # total providers and most recent upload
+    @providers = Organization.where(provider: true).count
+    @last_upload = Upload.order(created_at: :asc).last
+
+    # total number of MARC records & unique records (by 001 field)
+    @total_records = MarcRecord.count
+    @unique_records = MarcRecord.select('DISTINCT marc001').count
+
+    return if current_user.blank?
```

Comment:
I still think it might be nice to create a model/class to contain all the logic needed to fetch data to build this home page view. You could pass it the current user and whatever else it needs and then you'd be able to pass one instance of this new class to view with methods for fetching the data to you need.

Reply:
>  create a model/class to contain all the logic needed to fetch data 

👍 . I think w'll find some of the data fetching will be pretty expensive, so this will give us an easy place to cache it too.

Reply:
extracted this stuff to an `Overview` model and made that rely on pre-calculated statistics based on @corylown's advice. looking for more 👀 to see if anything else needs perf tuning; currently I can't ssh to prod to run ActiveRecord queries myself for some reason